### PR TITLE
Pytorch CNN Segmentation

### PR DESCRIPTION
**Pull request by shreyasingh1**
Pytorch 3D CNN model architecture for training on 3D voxels. Easily editable example notebook (found in experiments -> pytorch_model) illustrates the model's application on the 50 image Mouselight benchmarking data. Helper files are found in brainlit.utils.cnn_segmentation

### CODE REVIEW

**Comment by shreyasingh1** [comment]
*Still need to write tests, black files, and confirm that docs build

**Comment by shreyasingh1** [comment]
@tathey1 The PR now builds with CircleCI and Netlify. Code cov goes slightly down, only because I didn't write tests for functions that required "data_dir" as input. In the Netlify build, you can see the new refs in Utils->CNN Segmentation, and "make_masks" which is in Utils generally. 

Brief description of PR: in brainlit -> utils -> cnn_segmentation, I added two helper files called preprocess_cnn.py and performance_cnn.py that format data into torch objects and provide code to train, test, and optimize a 3D CNN. Example notebook of how to use these is in experiments -> pytorch_model. Also added a file in brainlit -> utils called make_masks.py that formalizes how to make 3D image masks from a series of SWC files. Didn't write tests for this file because Bresenham3D is tested elsewhere in brainlit (but can add here too if necessary).

Only standing question is that the "Return" arguments are formatted strangely in some of the functions in Netlify. Function docstrings seem to be formatted correctly and I tried a few different alternatives with no luck. Input parameters are formatted right, it's only the Return arguments that aren't for some.  Any guidance on how to fix?

File: brainlit/utils/make_masks.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    type_to_date,
+)
+
+
+def make_masks(data_dir):
+    """Swc to numpy mask
+
+    Arguments:
+        data_dir: direction to base data folder that download_benchmarking points to.
+        Should contain sample-tif-location and sample-swc-location
```
Comment (tathey1) [inline brainlit/utils/make_masks.py:21]: this newline is causing some issues in the docs page

**Review by tathey1 — CHANGES_REQUESTED** [review]
Two things:
- If the code coverage is not passing, can you please add in the description the rationale behind which methods you tested and which you did not (per our conversation).
- Please add type hints like [here](https://github.com/neurodata/brainlit/blob/c4dfa8b392e624b37e88a13195e4334806416d99/brainlit/algorithms/connect_fragments/viterbrain.py#L15)

After that, I will look at the code more closely. Thanks!
